src/Ships/P1MissileCorvette.c

```diff
@@ -174,22 +182,42 @@ bool P1MissileCorvetteSpecialTarget(Ship *ship,void *custom)
     {
         // have destroyed targets, so we are done
         spec->curTargetIndex = 0;
+        spec->volleyState = VOLLEY_BEGIN;
         return TRUE;
     }
 
+    if(spec->volleyState == VOLLEY_BEGIN)
+    {
+        spec->volleyState = VOLLEY_NOT_BEGIN;
+        //////////////////////
+        //Volley attack speech event!
+        //event num: COMM_MissleDest_VolleyAttack
+        //use battle chatter
+        if(ship->playerowner->playerIndex == universe.curPlayerIndex)
+        {
+            if (battleCanChatterAtThisTime(BCE_COMM_MissleDest_VolleyAttack, ship))
+            {
+                battleChatterAttempt(SOUND_EVENT_DEFAULT, BCE_COMM_MissleDest_VolleyAttack, ship, SOUND_EVENT_DEFAULT);
+            }
+        }
+        //////////////////////
+    }
+
     if (spec->curTargetIndex >= numShipsToTarget)
     {
         spec->curTargetIndex = 0;
     }
 
     numGuns = ship->gunInfo->numGuns;
     firedSomeMissiles = FALSE;
+    triedToFire=FALSE;
     for (i=0;i<numGuns;i++)
     {
         gunstatic = &shipstaticinfo->gunStaticInfo->gunstatics[i];
         if (gunstatic->guntype == GUN_MissileLauncher)
         {
             gun = &ship->gunInfo->guns[i];
+            triedToFire=TRUE;
             if (gunHasMissiles(gun))
             {
                 firedSomeMissiles = TRUE;
```

Comment:
[R] The logic in here could probably completely imitate the missile destroyer because the flying patterns of the missile corvette feel odd during volley fire, for some reason.

Reply:
Can't be done yet, unfortunately, at least not until the missile destroyer logic is fully understood: https://youtu.be/00o_PZQPHhw